### PR TITLE
🐛 Fix pallet-xcm benchmarks

### PR DESCRIPTION
**Pull request by JuaniRios**
## What?
- Fix pallet-xcm benchmarks

## Why?
- So we can have accurate weights for our runtimes 

## How?
- Add some janky patches to go around the shitty code of the pallet.
- Most of that shitty code is fixed in the latest version of pallet-xcm, but we cannot use that now till we upgrade substrate version

## Testing?
`just dry-run-benchmarks`

### CODE REVIEW

**Comment by JuaniRios** [comment]
* **#248** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/248?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JuaniRios and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: Cargo.toml [hunk 1]
```diff
@@ -237,4 +237,4 @@ polimec-runtime = { path = "runtimes/polimec" }
 asset-hub-polkadot-runtime = { git = 'https://github.com/polkadot-fellows/runtimes', default-features = false, rev = '59172d5fb8feea2646490618bead3b2b4fb06220' }
 polkadot-runtime-constants = { git = 'https://github.com/polkadot-fellows/runtimes', default-features = false, rev = '59172d5fb8feea2646490618bead3b2b4fb06220' }
 system-parachains-constants = { git = 'https://github.com/polkadot-fellows/runtimes', default-features = false, rev = '59172d5fb8feea2646490618bead3b2b4fb06220' }
-polkadot-runtime = { git = 'https://github.com/polkadot-fellows/runtimes', default-features = false, rev = '59172d5fb8feea2646490618bead3b2b4fb06220' }
+polkadot-runtime = { git = 'https://github.com/polkadot-fellows/runtimes', default-features = false, rev = '59172d5fb8feea2646490618bead3b2b4fb06220' }
```
Comment (vstam1) [inline Cargo.toml:240]: Nit: Remove this new line

File: runtimes/polimec/src/lib.rs [hunk 2]
```diff
@@ -429,9 +429,35 @@ impl cumulus_pallet_aura_ext::Config for Runtime {}
 parameter_types! {
 	pub const RelayOrigin: AggregateMessageOrigin = AggregateMessageOrigin::Parent;
 }
+pub struct MockedChannelInfo;
```
Comment (vstam1) [inline runtimes/polimec/src/lib.rs:432]: Can you add a todo comment to this struct why we need it and when we can remove it?
Might also be good to make a linear issue for all TODOs we need to do for next version upgrade. Haven't done it myself but just though of it.

File: runtimes/polimec/src/lib.rs [hunk 3]
```diff
@@ -1195,17 +1221,27 @@ impl_runtime_apis! {
 			use cumulus_pallet_session_benchmarking::Pallet as SessionBench;
 			use xcm::latest::prelude::*;
 			use pallet_xcm::benchmarking::Pallet as PalletXcmExtrinsiscsBenchmark;
-			/// TODO: Update these benchmarks once we enable PLMC Teleportation
+			/// TODO: Update these janky implementations once we upgrade pallet_xcm. The jankyness is due to badly written pallet-xcm code
```
Comment (vstam1) [inline runtimes/polimec/src/lib.rs:1224]: Can you keep the old TODO as well? And maybe remove the comment on badly written as its an open source repo 🤣

File: runtimes/politest/src/lib.rs [hunk 4]
```diff
@@ -456,8 +456,36 @@ parameter_types! {
 	pub const RelayOrigin: AggregateMessageOrigin = AggregateMessageOrigin::Parent;
 }
 
+pub struct MockedChannelInfo;
```
Comment (vstam1) [inline runtimes/politest/src/lib.rs:459]: Same as other comment

File: runtimes/politest/src/lib.rs [hunk 5]
```diff
@@ -1401,17 +1429,27 @@ impl_runtime_apis! {
 			use cumulus_pallet_session_benchmarking::Pallet as SessionBench;
 			use xcm::latest::prelude::*;
 			use pallet_xcm::benchmarking::Pallet as PalletXcmExtrinsiscsBenchmark;
-			/// TODO: Update these benchmarks once we enable PLMC Teleportation
+			/// TODO: Update these janky implementations once we upgrade pallet_xcm. The jankyness is due to badly written pallet-xcm code
```
Comment (vstam1) [inline runtimes/politest/src/lib.rs:1432]: Same

File: runtimes/polimec/src/lib.rs [hunk 6]
```diff
@@ -1195,17 +1221,27 @@ impl_runtime_apis! {
...
+						RuntimeOrigin::root(),
+						Box::new(crate::xcm_config::AssetHubLocation::get()),
+						xcm::prelude::XCM_VERSION
+					).unwrap();
 					Some((
 						MultiAsset {
-							fun: Fungible(EXISTENTIAL_DEPOSIT),
-							id: Concrete(Parent.into())
+							fun: Fungible(ExistentialDeposit::get()),
+							id: Concrete(Here.into())
```
Comment (vstam1) [inline runtimes/polimec/src/lib.rs:1244]: Why should this be Here? As Here describes PLMC in this context, right? Instead of Dot?

**Review by vstam1 — CHANGES_REQUESTED** [review]
Nice that you got the pallet-xcm benchmarks to work. If I understand it correctly, the two problems we run into are:
- XCM version is not set? Is this fixed in future versions, and how?
- The channel info is normally not properly mocked? How do other chains fix this, as this should be a general problem right?

File: runtimes/polimec/src/lib.rs (hunk 6 above)
Comment (JuaniRios) [reply runtimes/polimec/src/lib.rs:1244]: Benchmarks use Balances pallet to fund the accounts before the reserve transfer, so we only have the choice of specifying our native asset which was one of the bugs solved in this PR

File: Cargo.toml [hunk 7]
```diff
@@ -237,4 +237,3 @@ polimec-runtime = { path = "runtimes/polimec" }
 asset-hub-polkadot-runtime = { git = 'https://github.com/polkadot-fellows/runtimes', default-features = false, rev = '59172d5fb8feea2646490618bead3b2b4fb06220' }
 polkadot-runtime-constants = { git = 'https://github.com/polkadot-fellows/runtimes', default-features = false, rev = '59172d5fb8feea2646490618bead3b2b4fb06220' }
 system-parachains-constants = { git = 'https://github.com/polkadot-fellows/runtimes', default-features = false, rev = '59172d5fb8feea2646490618bead3b2b4fb06220' }
-polkadot-runtime = { git = 'https://github.com/polkadot-fellows/runtimes', default-features = false, rev = '59172d5fb8feea2646490618bead3b2b4fb06220' }
```
Comment (vstam1) [inline Cargo.toml:240]: Wait, is the dep now removed? Should still keep it for integration tests, right?

File: runtimes/polimec/src/lib.rs (hunk 6 above)
Comment (vstam1) [reply runtimes/polimec/src/lib.rs:1249]: Hmmm interesting that it works with Here, as we should not be able to reserve_based transfer PLMC.

**Review by vstam1 — APPROVED** [review]
After comments, I approve

File: Cargo.toml (hunk 7 above)
Comment (JuaniRios) [reply Cargo.toml:240]: Thought your comment was to delete this line?

Comment (vstam1) [reply Cargo.toml:240]: I think in the previous version is added a new line after this line. So was a nit to remove the new line. But we should probably still include this line, else integration tests won't compile

**Pull request merged**